app/lib/dns.server.ts

```diff
@@ -157,8 +168,29 @@ export const getChangeStatus = async (changeId: string) => {
   }
 };
 
-const isNameValid = (name: string) => {
-  return /^[a-z0-9-]+.[a-z0-9-]+.[a-z]+.[a-z]+\.?$/.test(name);
+/* Domain name rules
+1. Domain name pattern should be [name].[username].rootDomain.com
+2. Domain name can contain only alphanumerical characters, '-', and '_'
+3. Domain name should not start or end with -
+4. Domain name cannot contain multiple consecutive '-' or '_'
+5. Domain name can contain uppercase in UI but it is converted to lowercase before validation */
+export const isNameValid = (name: string, username: string) => {
+  const rootDomain = process.env.ROOT_DOMAIN!;
+  let subdomain: string = '';
```

Comment:
There's no value in doing this.  Define your variable where you'll use it, and don't set it equal to a string you'll never use (empty string).